### ISSUE TITLE
Dynamic parameter names

### ISSUE THREAD

**Issue by kevinresol** [haxetink/tink_web#76]
```haxe
@:params(bar('bar-$_') in query)
function get(bar:DynamicAccess<String>);
```

and a request with `?bar-foo=a&bar-baz=b` would be translated to `{foo:'a', baz:'b'}` in Haxe.

Any thoughts?

**Comment by back2dos** [comment]
Well, the main problem here is parsing `DynamicAccess`, which is currently unsupported. It'll require restructuring [how the data is stored in the parser](https://github.com/haxetink/tink_querystring/blob/f1710c07d8ed5d01521c3db10288f7c9733c0411/src/tink/querystring/Parser.hx#L12), because getting maps from a flat key-value-list is not *that* easy. Especially if you consider more complex cases like `DynamicAcces<String, { ?foo:String, ?bar:Int }>`.

**Comment by kevinresol** [comment]
Query string is only part of the problem. I also want this to work with request headers.  (Thanks to aws)

**Comment by back2dos** [comment]
Yeah, but headers are basically a key-value list too ... both get parsed via tink_querystring, which despite its name is not *that* query string specific ^^
